primitives/core/src/crypto.rs

```diff
@@ -647,6 +647,28 @@ impl<'de> serde::Deserialize<'de> for AccountId32 {
 	}
 }
 
+#[cfg(feature = "std")]
+impl sp_std::str::FromStr for AccountId32 {
+	type Err = &'static str;
+
+	fn from_str(s: &str) -> Result<Self, Self::Err> {
+		if &s[..2] == "0x" && s.len() == 66 {
+			// try and parse 0x hex string
+			let mut bytes = [0u8; 32];
+			hex::decode_to_slice(&s[2..], &mut bytes).map_err(|_| "invalid hex address.")?;
+			Ok(bytes.into())
+		} else if s.len() == 64 {
+			// or string without 0x
+			let mut bytes = [0u8; 32];
+			hex::decode_to_slice(s, &mut bytes).map_err(|_| "invalid hex address.")?;
+			Ok(bytes.into())
```

Comment:
```suggestion
			hex::decode_to_slice(s, &mut bytes).map_err(|_| "invalid hex address.").map(Self)
```

Reply:
I think you wanted to suggest using `hex::decode`, because `decode_to_slice ` returns `Result<(), ...>` and cannot `.map()` on it. 

Even so, `AccountId32` has `TryFrom<&[u8]>` and then I'd have to `expect("why this is valie")`. I think using `decode` is better to directly use `From<[u8; 32]>`.

Reply:
Or 

```
hex::decode_to_slice(s, &mut bytes)
				.map_err(|_| "invalid hex address.")
				.map(|_| Self::from(bytes))
```